### PR TITLE
Apply 2024.1 backports from 2023.1

### PR DESCRIPTION
**Pull request by seunghun1ee**
Applied following commits
Commit | Change-Id | Message
58c21f9662 Icd95a86e6443a478708900324a68a2855d6f3837 Conditionally deploy caso
ba8746f6af None Backport fixes
85341c2dad Ib81ec443bbb14d494d7c81dff617b371a1b68c52 Add role to deploy cASO

Followings were not applied
Commit | Change-Id | Message
8b7d17ec99 None Remove nova_client.endpoint_type conditional # Magnum change that is no longer used
c99645d6b6 None Rename ansible var to magnum_capi_helm_driver_enabled # Magnum change that is no longer used
7df3b8fa57 None Make task name clearer # Magnum change that is no longer used
61c4e56ae9 None Fix variable name # Magnum change that is no longer used
01df7a18ce None Set nova-client endpoint_type appropriately # Magnum change that is no longer used
127083d3f2 None Remove unnecessary kubeconfig path override # Magnum change that is no longer used
9de4158b12 None Rename config section to match driver change # Magnum change that is no longer used
190a99a1d9 None Rename capi-helm-driver var # Magnum change that is no longer used
907ecca68c None Remove duplicate tasks # Magnum change that is no longer used
639605349d None Change kolla_container to kolla_docker # No need to change
34b042fa7b None Reinstate missing config snippet # Makes no changes
3c8040133e None Revert "Fix task ordering" # Pair of commit and revert
9b7fb2bc19 None Fix task ordering # Pair of commit and revert
48bcaa8e73 None Revert "Fix nova DB password rotation" # Pair of commit and revert
9af6e421f2 Ibdbab1b25fb9418dc1bdda19a092c4d990ebee5b Fix nova DB password rotation # Pair of commit and revert
a9728c6581 None remove whitespace # Overwrites more recent change
cce0b69abe None config group has changed upstream # Magnum change that is no longer used
8d3bebb12a None Update ansible/roles/magnum/templates/magnum.conf.j2 # Overwrites more recent change
f8c48affb4 None Changes for cluster-api magnum driver # Overwrites more recent change

### CODE REVIEW

File: ansible/roles/grafana/tasks/post_config.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 - name: Remove old grafana docker volume
   become: true
-  kolla_container:
```
Comment (Alex-Welsh) [inline ansible/roles/grafana/tasks/post_config.yml:18]: Not sure about this change

Comment (seunghun1ee) [reply ansible/roles/grafana/tasks/post_config.yml:18]: Yes I checked the history and I can see that we can use kolla_container again.

File: ansible/roles/caso/tasks/config.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  when:
+    - inventory_hostname in groups[item.value.group]
+    - item.value.enabled | bool
+  with_dict: "{{ caso_services }}"
+  notify:
+    - Restart caso container
+
+- name: Check caso containers
+  become: true
+  kolla_docker:
```
Comment (markgoddard) [inline ansible/roles/caso/tasks/config.yml:76]: Needs `container_engine: "{{ kolla_container_engine }}"`?

File: ansible/roles/caso/tasks/pull.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+- name: Pulling caso images
+  become: true
+  kolla_docker:
+    action: "pull_image"
```
Comment (markgoddard) [inline ansible/roles/caso/tasks/pull.yml:5]: container_engine: "{{ kolla_container_engine }}"

File: ansible/roles/caso/handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Restart caso container
+  vars:
+    service_name: "caso"
+    service: "{{ caso_services[service_name] }}"
+    config_json: "{{ caso_config_jsons.results|selectattr('item.key', 'equalto', service_name)|first }}"
+    caso_container: "{{ check_caso_containers.results|selectattr('item.key', 'equalto', service_name)|first }}"
+  become: true
+  kolla_docker:
```
Comment (markgoddard) [inline ansible/roles/caso/handlers/main.yml:9]:
```suggestion
  kolla_container:
```

File: ansible/roles/caso/tasks/config.yml (hunk 2 above)
Comment (markgoddard) [inline ansible/roles/caso/tasks/config.yml:76]:
```suggestion
  kolla_container:
```

File: ansible/roles/caso/tasks/pull.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
+- name: Pulling caso images
+  become: true
+  kolla_docker:
```
Comment (markgoddard) [inline ansible/roles/caso/tasks/pull.yml:4]:
```suggestion
  kolla_container:
```

File: ansible/roles/magnum/tasks/config.yml [hunk 6]
```diff
@@ -58,6 +58,10 @@
   when:
     - magnum_kubeconfig_file.stat.exists
 
+- name: Enable magnum cluster-api helm driver if kubeconfig exists
+  set_fact:
+    magnum_capi_helm_driver_enabled: "{{ magnum_kubeconfig_file.stat.exists | bool }}"
```
Comment (markgoddard) [inline ansible/roles/magnum/tasks/config.yml:63]: This variable is unused (it was also unused in antelope...)

**Review by markgoddard — CHANGES_REQUESTED** [review]
If we remove the magnum change, then only caso changes remain. Could we remove the other commits from the history to clean up?

**Comment by seunghun1ee** [comment]
Why do we remove magnum changes?

**Comment by markgoddard** [comment]
> Why do we remove magnum changes?

Perhaps @scrungus or @sd109 can confirm, but I can't see the `magnum_capi_helm_driver_enabled` variable being used anywhere.

**Comment by scrungus** [comment]
this variable is unused, we now use `magnum_kubeconfig_file_path` to enable magnum configuration for the new driver, which is autodetected - see:
https://github.com/stackhpc/kolla-ansible/blob/d1335a5b52f6f871c61b92c0325bd8fd2b554725/ansible/roles/magnum/templates/magnum.conf.j2#L155C7-L158
and:
https://github.com/stackhpc/kolla-ansible/blob/d1335a5b52f6f871c61b92c0325bd8fd2b554725/ansible/roles/magnum/tasks/config.yml#L34-L59

**Comment by seunghun1ee** [comment]
OK. I'll drop magnum changes and make caso changes that @markgoddard suggested.

**Review by markgoddard — APPROVED** [review]

**Comment by markgoddard** [comment]
Squashed into one commit.

**Pull request merged**